### PR TITLE
APPS-1102 Improve documentation about outputting DX files

### PR DESCRIPTION
**Pull request by mhrvol**

### CODE REVIEW

**Review by commandlinegirl — CHANGES_REQUESTED** [review]
Looks good, please improve formatting , e.g. add space before `{` or `<<<`.

**Comment by commandlinegirl** [comment]
Adding `minor` label so that all tests are skipped for this doc update.

File: doc/ExpertOptions.md [hunk 1]
```diff
@@ -579,6 +579,51 @@ workflow math {
 
 Currently, dxCompiler does not support this feature. However, there is a [suggestion](MissingCallArguments.md) for limited support.
 
+# DNAnexus files as outputs
+
+User can work with DNAnexus files inside WDL tasks or workflow using dx commands and create tasks or workflows that output DNAnexus files. This can be used to generate multiple outputs without storing all of them on local filesystem.
```
Comment (emiloslavsky) [inline doc/ExpertOptions.md]: I think this only applies to tasks, not workflows.

File: doc/ExpertOptions.md [hunk 2]
```diff
@@ -579,6 +579,51 @@ workflow math {
...
+task find_fastq_from_previous{
+  input{
+    String sample
+    String old_fastq_folder
+    String R1_or_R2
+  }
+  command<<<
+    project=$DX_PROJECT_CONTEXT_ID
+    folder=$project://~{old_fastq_folder}
+    fq=$(dx find data --name "{sample}{R1_or_R2}.fastq.gz" --path ${folder} --norecurse --brief)
```
Comment (emiloslavsky) [inline doc/ExpertOptions.md]: do you need to use ~{sample} (with tilda)
I'd also simplify this example to remove R1_or_R2

File: doc/ExpertOptions.md [hunk 3]
```diff
@@ -579,6 +579,51 @@ workflow math {
 
 Currently, dxCompiler does not support this feature. However, there is a [suggestion](MissingCallArguments.md) for limited support.
 
+# DNAnexus files as outputs
+
+User can work with DNAnexus files inside WDL tasks or workflow using dx commands and create tasks or workflows that output DNAnexus files. This can be used to generate multiple outputs without storing all of them on local filesystem.
+
+## Example 1: DNAnexus files as outputs
+```wdl
+task find_fastq_from_previous{
```
Comment (emiloslavsky) [inline doc/ExpertOptions.md]:
```suggestion
task find_fastq_from_previous {
```

File: doc/ExpertOptions.md [hunk 4]
```diff
@@ -579,6 +579,51 @@ workflow math {
...
+  input{
```
Comment (emiloslavsky) [inline doc/ExpertOptions.md]:
```suggestion
  input {
```

File: doc/ExpertOptions.md [hunk 5]
```diff
@@ -579,6 +579,51 @@ workflow math {
...
+  command<<<
+    project=$DX_PROJECT_CONTEXT_ID
+    folder=$project://~{old_fastq_folder}
+    fq=$(dx find data --name "{sample}{R1_or_R2}.fastq.gz" --path ${folder} --norecurse --brief)
+    for file_id in $fq
+    do
+    echo "dx://${file_id}"
+    done
+  >>>
+  output{
```
Comment (emiloslavsky) [inline doc/ExpertOptions.md]:
```suggestion
  output {
```

File: doc/ExpertOptions.md (hunk 1 above)
Comment (emiloslavsky) [reply doc/ExpertOptions.md]: this only works without docker container.

File: doc/ExpertOptions.md [hunk 6]
```diff
@@ -579,6 +579,51 @@ workflow math {
...
+    Array[File] fq = read_lines(stdout())
+  }
+}
+```
+
+## Example 2: Uploading local file to DNAnexus
+
+```wdl
+task upload_and_delete{
```
Comment (emiloslavsky) [inline doc/ExpertOptions.md]: Idea is to show prudent use of disk space.
A nicer illustration would involve unzipping an array of files, where after unzipping each input file, we tag and upload the unzipped resulting file.

**Review by emiloslavsky — CHANGES_REQUESTED** [review]

File: doc/ExpertOptions.md [hunk 7]
```diff
@@ -579,6 +579,54 @@ workflow math {
...
+            rm "${f_unzipped}"
+            echo "dx://${DX_WORKSPACE_ID}:${file_id}"
+        done
+    >>>
+    output {
+        Array[File] files_out = read_lines(stdout())
+    }
+}
+```
+
```
Comment (emiloslavsky) [inline doc/ExpertOptions.md:629]: these examples need to be added to integration tests to make sure the always work.

**Review by emiloslavsky — CHANGES_REQUESTED** [review]

File: doc/ExpertOptions.md (hunk 7 above)
Comment (mhrvol) [reply doc/ExpertOptions.md:629]: Added integration test.

Comment (emiloslavsky) [reply doc/ExpertOptions.md:629]: what about the adding first example to the integration tests?
The idea is that examples always pass our tests.

Comment (mhrvol) [reply doc/ExpertOptions.md:629]: Added the second example. I had to change our test script to deal with dx files as it was impossible to check if the outputs were correct.

**Review by emiloslavsky — DISMISSED** [review]

**Pull request merged**